### PR TITLE
Add Anvil Native keyboard

### PR DESCRIPTION
**Pull request by ebastler**
This PR adds the Anvil Native keyboard, both ISO and ANSI version.

## Description

It uses per key RGB with IS31 drivers, and features some custom LED code, which is for the most part confined into one keymap, with the exception of a custom LED animation that is present in all keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Add support for Anvil Native

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/teleport/native/ansi/ansi.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+led_config_t g_led_config = { {
```
Comment (drashna) [inline keyboards/teleport/native/ansi/ansi.c:19]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
led_config_t g_led_config = { {
```

File: keyboards/teleport/native/config.h [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+/* Since the typing heatmap is broken on duplex matrix boards, force it to "slim" mode */
+#ifdef ENABLE_RGB_MATRIX_TYPING_HEATMAP
+    #define RGB_MATRIX_TYPING_HEATMAP_SLIM
+#endif
+
+/* Allow VIA to edit lighting */
+#ifdef VIA_ENABLE
+    #define VIA_QMK_RGBLIGHT_ENABLE
+#endif
```
Comment (drashna) [inline keyboards/teleport/native/config.h:114]: This is no longer needed.  Via handles both the rgb light and rgb matrix systems properly now (eg, the v3)
```suggestion
```

File: keyboards/teleport/native/iso/iso.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+led_config_t g_led_config = { {
```
Comment (drashna) [inline keyboards/teleport/native/iso/iso.c:19]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
led_config_t g_led_config = { {
```

File: keyboards/teleport/native/iso/iso.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    {  2,64 }, { 21,64 }, { 39,64 },                                  { 95,64 },                                  {149,64 }, {164,64 }, {179,64 }, {194,64 }, {209,64 }, {224,64 }
+}, {
+    // LED Index to Flag
+    1, 4, 4, 4, 4, 1, 1, 1, 1, 4, 4, 4, 4, 1, 1, 1,
+    4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1, 1,
+    1,    4,12, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 1,
+    9,   12,12,12, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1,
+    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1, 4, 1,
+    1, 9, 1,          4,          1, 1, 1, 4, 4, 4,
+} };
```
Comment (drashna) [inline keyboards/teleport/native/iso/iso.c:49]:
```suggestion
} };
#endif
```

File: keyboards/teleport/native/native.c [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+#include "quantum.h"
+
+/* This board has !SDB of the is31 wired to D2. Set high to enable */
+void keyboard_post_init_user(void) {
+    setPinOutput(B9);
+    writePinHigh(B9);
+}
+
+const is31_led PROGMEM g_is31_leds[RGB_MATRIX_LED_COUNT] = {
```
Comment (drashna) [inline keyboards/teleport/native/native.c:25]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
const is31_led PROGMEM g_is31_leds[RGB_MATRIX_LED_COUNT] = {
```

File: keyboards/teleport/native/native.c [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    {1, F_2,  E_2,  D_2},
+    {1, F_3,  E_3,  D_3},
+    {1, F_7,  E_7,  D_7},
+    {1, F_11, E_11, D_11},
+    {1, F_12, E_12, D_12},
+    {1, F_13, E_13, D_13},
+    {1, F_14, E_14, D_14},
+    {1, F_15, E_15, D_15},
+    {1, F_16, E_16, D_16}
+};
```
Comment (drashna) [inline keyboards/teleport/native/native.c:122]:
```suggestion
};
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/teleport/native/rules.mk [hunk 7]
```diff
@@ -0,0 +1,5 @@
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = IS31FL3733
+RGB_MATRIX_CUSTOM_KB = yes
+
+DEFAULT_FOLDER = /teleport/native/iso
```
Comment (drashna) [inline keyboards/teleport/native/rules.mk:5]:
```suggestion
DEFAULT_FOLDER = teleport/native/iso
```

Comment (tzarc) [reply keyboards/teleport/native/rules.mk:5]: Shouldn't the whole thing be under `anvil`, not `teleport`?

Comment (ebastler) [reply keyboards/teleport/native/rules.mk:5]: Anvil is the gaming brand by teleport, so we opted to keep them in the same folder, and with the same vid. It reports as Anvil to the PC, but everything else is shared with other teleport designs.

**Review by tzarc — APPROVED** [review]

**Pull request merged**